javascript/react/src/components/ArqueroDetailsList/renderers/ValidationCell.tsx

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { FieldError, ValidationTestResult } from '@datashaper/schema'
+import { ErrorCode } from '@datashaper/schema'
+import type { ITooltipHostStyles } from '@fluentui/react/lib/Tooltip'
+import { TooltipHost } from '@fluentui/react/lib/Tooltip'
+import { useId } from '@fluentui/react-hooks'
+import { memo } from 'react'
+
+import { getValue } from '../ArqueroDetailsList.utils.js'
+import { useIconProps } from '../hooks/useIconProps.js'
+import { useValidationIconProps } from '../hooks/useValidationIconProps.js'
+import { useTextAlignStyle } from './hooks.js'
+import type { FormattedCellProps } from './types.js'
+import {
+	Container,
+	LeftIcon,
+	ToolTipContainer,
+	ValueContainer,
+} from './ValidationCell.styles.js'
+
+/**
+ * Basic rendering of text values.
+ */
+export const ValidationCell: React.FC<FormattedCellProps> = memo(
+	function TextCell({ item, column, textAlign = 'left', validationResult }) {
+		const value = getValue(item, column)
+		const style = useTextAlignStyle(textAlign)
+		const hostStyles: Partial<ITooltipHostStyles> = {
+			root: { display: 'inline-block' },
+		}
+		const tooltipId = useId('tooltip')
+		const calloutProps = { gapSpace: 0 }
+
+		const iconProps = useIconProps(validationResult)
+		const validationIconProps = useValidationIconProps(
+			iconProps,
+			validationResult,
+		)
+
+		return (
+			<Container style={style}>
+				{validationResult !== undefined &&
+					validationResult.errors.map((e: FieldError, index: number) => {
```

Comment:
I would move all of this logic out to a hook so there is not a bunch of if statements in the jsx.